vscode/src/test/launchNbcode.ts

```diff
@@ -0,0 +1,73 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import * as os from 'os';
+import { env } from 'process';
+import { ChildProcessByStdio, spawn } from 'child_process';
+import { Readable } from 'stream';
+
+const findNbcode = (extensionPath: string): string => {
+    let nbcode = os.platform() === 'win32' ?
+        os.arch() === 'x64' ? 'nbcode64.exe' : 'nbcode.exe'
+        : 'nbcode.sh';
+    let nbcodePath = path.join(extensionPath, "nbcode", "bin", nbcode);
+
+    let nbcodePerm = fs.statSync(nbcodePath);
+    if (!nbcodePerm.isFile()) {
+        throw `Cannot execute ${nbcodePath}`;
+    }
+    if (os.platform() !== 'win32') {
+        fs.chmodSync(path.join(extensionPath, "nbcode", "bin", nbcode), "744");
+        fs.chmodSync(path.join(extensionPath, "nbcode", "platform", "lib", "nbexec.sh"), "744");
+        fs.chmodSync(path.join(extensionPath, "nbcode", "java", "maven", "bin", "mvn.sh"), "744");
+    }
+    return nbcodePath;
+}
```

Comment:
This seems like a duplicate of the function in *lsp/utils.ts* possibly a remnant of the merge of the testing branch.

Reply:
Yes maybe due to that, removed duplicate

Reply:
Oh ok. Understood. Thanks.